client/setec/store.go

```diff
@@ -352,19 +352,35 @@ func (s *Store) lookupSecretInternal(ctx context.Context, name string) (Secret,
 		ctx, cancel = context.WithTimeout(s.ctx, 5*time.Minute)
 		defer cancel()
 	}
-	sv, err := s.client.Get(ctx, name)
-	if err != nil {
-		return nil, fmt.Errorf("lookup %q: %w", name, err)
-	}
 
-	s.active.Lock()
-	defer s.active.Unlock()
-	s.active.m[name] = &cachedSecret{Secret: sv, LastAccess: s.timeNow().Unix()}
-	if err := s.flushCacheLocked(); err != nil {
-		s.logf("WARNING: error flushing cache: %v", err)
+	// When lookups are enabled, it is possible a bunch of goroutines may race
+	// for the right to grab and cache a given secret, so singleflight the
+	// lookup for each secret under its own marker. The "lookup:" prefix here
+	// ensures we don't collide with the "poll" label used by periodic updates.
+	ch := s.single.DoChan("lookup:"+name, func() (any, error) {
+		sv, err := s.client.Get(ctx, name)
```

Comment:
this ctx might be a different person's context.

You probably want to make this context be disconnected and just be like a few seconds.

the select on 375 will handle cancel/timeout of the caller's actual context?

Reply:
Oh good point, I had originally done the context setup inside the callback, but then talked myself out of it. But that's a good point. Let me clean that up.

Reply:
Switched this one to use plain Do, since the context handling is already wired up in all the things this calls.